### PR TITLE
fix(components): [button] :not overrides the disabled style

### LINKED ISSUE #8013: [Bug Report] [Component] [button] el-button在v2.2.0及以上版本出现样式问题
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.33`
- Element Plus Version: `2.2.0`
- Browser / OS: `Chrome/75.0.3770.100`
- Build Tool: `Vite`

### Related Component

- `el-button`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24gfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAg5Zyo5L2O54mI5pys5rWP6KeI5Zmo77yIY2hyb21lNzXvvInnrYnniYjmnKzvvIxlbGVtZW50LXBsdXMgdjIuMi4w5Y+K5Lul5LiK54mI5pys5oyJ6ZKu55qE5Lyq57G75qC35byP5LiN55Sf5pWI77yM5a+86Ie05oyJ6ZKu6IOM5pmv54Gw6Imy5oiW5LiN5Y+v6KeB77yI55u05o6l5L2/55SoY2hyb21lNzXmiJbogIVRUea1j+iniOWZqOi/m+WFpWVsZW1lbnQtcGx1c+eahOWumOe9keS5n+WPr+WkjeeOsO+8iVxuICA8ZWwtYnV0dG9uIHR5cGU9XCJwcmltYXJ5XCI+XG4gICAg5rWL6K+V5oyJ6ZKuXG4gIDwvZWwtYnV0dG9uPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==)

### Steps to reproduce

直接使用Chrome75版本或者QQ浏览器进入element-plus官网即可复现

### What is Expected?

按钮显示正常

### What is actually happening?

按钮背景为灰色或直接不可见

### Additional comments

查看代码发现是伪类不生效

### PR DESCRIPTION
**Pull request by btea**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

Before: 
image

After:
image

### CODE REVIEW

File: packages/theme-chalk/src/button.scss [hunk 1]
```diff
@@ -40,7 +40,9 @@ $button-icon-span-gap: map.merge(
   vertical-align: middle;
   -webkit-appearance: none;
 
-  &:not(.is-text):not(.is-link):not(.#{$namespace}-button--text) {
+  &:not(.is-text),
```
Comment (tolking) [inline packages/theme-chalk/src/button.scss:43]: this has the same effect with remove `:not`, maybe we can consider remove `:not()`.

Comment (btea) [reply packages/theme-chalk/src/button.scss:43]: If  remove `:not()`, the basic usage seems to lose the style.

File: packages/theme-chalk/src/button.scss [hunk 2]
```diff
@@ -127,6 +127,9 @@ $button-icon-span-gap: map.merge(
       background-image: none;
       background-color: getCssVar('button', 'disabled', 'bg-color');
       border-color: getCssVar('button', 'disabled', 'border-color');
+      &:not(.is-text) {
```
Comment (tolking) [inline packages/theme-chalk/src/button.scss:130]: it overrides the border color of Disabled Button

File: packages/theme-chalk/src/button.scss (hunk 1 above)
Comment (tolking) [reply packages/theme-chalk/src/button.scss]:
```diff
- &:not(.is-text):not(.is-link):not(.#{$namespace}-button--text) { // line 43

- } // line 68
```

I tried it and didn't find the error you said

Comment (btea) [reply packages/theme-chalk/src/button.scss]: I seem to have misunderstood what you meant. Did you mean to delete lines 43 to 68?

Comment (tolking) [reply packages/theme-chalk/src/button.scss]: just 43 and 68

Comment (btea) [reply packages/theme-chalk/src/button.scss]: I tried it and it looks very good 👍

**Review by tolking — APPROVED** [review]

**Review by jw-foss — APPROVED** [review]

**Pull request merged**